src/batch/src/executor/join/hash_join.rs

```diff
@@ -907,7 +907,6 @@ impl<K: HashKey> HashJoinExecutor<K> {
                 if !visible {
                     continue;
                 }
-                non_equi_state.found_matched = false;
```

Comment:
🤔  should we do it?

Reply:
Yes, it should only be modified inside `remove_duplicate_rows_for_left_outer_join`.